src/lib/mdns/Resolver.h

```diff
@@ -44,9 +44,12 @@ struct ResolvedNodeData
     }
 
     PeerId mPeerId;
-    Inet::InterfaceId mInterfaceId;
-    Inet::IPAddress mAddress;
-    uint16_t mPort;
+    Inet::IPAddress mAddress         = Inet::IPAddress::Any;
+    Inet::InterfaceId mInterfaceId   = INET_NULL_INTERFACEID;
+    uint16_t mPort                   = 0;
+    bool mSupportsTcp                = false;
+    uint32_t mMrpRetryIntervalIdle   = 0;
+    uint32_t mMrpRetryIntervalActive = 0;
```

Comment:
Should those be `Optional<uint32_t>` so we can clearly tell apart "not advertised" from "advertised as 0"?  I see nothing in the spec preventing 0 being advertised as a valid value....

Reply:
Hmm, I don't like the memory overhead `Optional` generates (maybe we should have a specialization of Optional for unsigned integers that don't use the most significant bit?), but I can definitely use some `kUndefined` constant. Imho it doesn't really make sense to advertise CRI equal to 0 and how it should be handled differently than the lack of CRI, agree that technically it's possible.

Reply:
Having some sort of kUndefined constants would be fine especially if we then have accessors that either return `Optional` or at least let you ask whether the value was set, so consumers don't have to deal with the undefined values themselves...  This is especially ok if we do range-enforcement in this code, because then we can just pick any value > `3600000` as our "undefined" value.

And please feel free to raise a spec issue about 0 if we think that part of the spec is just technically wrong!